rts/motoko-rts/src/gc/incremental/roots.rs

```diff
@@ -1,75 +1,65 @@
-use crate::{types::Value, visitor::pointer_to_dynamic_heap};
+use motoko_rts_macros::ic_mem_fn;
+
+use crate::{types::Value, visitor::is_non_null_pointer_field};
+
+/// An array referring to the static program variables, being
+/// - All canister variables.
+/// - Pooled shared objects.
+/// The array constitutes a GC root that is reinitialized on each canister upgrade.
+/// The scalar sentinel denotes an uninitialized root.
+#[cfg(feature = "ic")]
+static mut STATIC_VARIABLES: Value = Value::from_scalar(0);
```

Comment:
Perhaps use a tagged scalar here. Probably not crucial since I don't think this will ever be seen in the heap.

Reply:
I could change this to `NULL_POINTER`. This is only a sentinel value (that is hopefully never read).